File: multi_agent/projects/im-mobile/lib/services/pin_service.dart
import 'package:dio/dio.dart';
import '../models/conversation_pin.dart';

class PinService {
  static final _dio = Dio(BaseOptions(
    baseUrl: '/api/conversations/pin',
    connectTimeout: const Duration(seconds: 10),
    receiveTimeout: const Duration(seconds: 10),
  ));

  static Future<PinConversationResponse> pin(PinConversationRequest request) async {
    try {
      final response = await _dio.post('/', data: request.toJson());
      return PinConversationResponse.fromJson(response.data);
    } on DioException catch (e) {
      return PinConversationResponse(success: false, message: e.message ?? 'Error');
    }
  }

  static Future<PinConversationResponse> unpin(int conversationId) async {
    try {
      final response = await _dio.delete('/$conversationId');
      return PinConversationResponse.fromJson(response.data);
    } on DioException catch (e) {
      return PinConversationResponse(success: false, message: e.message ?? 'Error');
    }
  }

  static Future<List<ConversationPin>> getPinned() async {
    try {
      final response = await _dio.get('/');
      return (response.data as List).map((e) => ConversationPin.fromJson(e)).toList();
    } on DioException {
      return [];
    }
  }

  static Future<PinConversationResponse> reorder(List<int> ids) async {
    try {
      final response = await _dio.put('/reorder', data: {'conversationIds': ids});
      return PinConversationResponse.fromJson(response.data);
    } on DioException catch (e) {
      return PinConversationResponse(success: false, message: e.message ?? 'Error');
    }
  }

  static Future<bool> isPinned(int conversationId) async {
    try {
      final response = await _dio.get('/$conversationId/status');
      return response.data['pinned'] ?? false;
    } on DioException {
      return false;
    }
  }
}

File: im-desktop/src/services/pin-service.ts
import axios from 'axios';
import { PinConversationRequest, PinConversationResponse, PinnedConversation } from '../types/conversation-pin';
import { useAuthStore } from '../stores/auth-store';

const api = axios.create({ baseURL: '/api/conversations/pin', timeout: 10000 });

api.interceptors.request.use((config) => {
  const user = useAuthStore.getState().user;
  if (user) config.headers['X-User-Id'] = user.id;
  return config;
});

export class PinService {
  static async pin(request: PinConversationRequest): Promise<PinConversationResponse> {
    const response = await api.post<PinConversationResponse>('/', request);
    return response.data;
  }

  static async unpin(conversationId: number): Promise<PinConversationResponse> {
    const response = await api.delete<PinConversationResponse>(`/${conversationId}`);
    return response.data;
  }

  static async getPinned(): Promise<PinnedConversation[]> {
    const response = await api.get<PinnedConversation[]>('/');
    return response.data;
  }

  static async reorder(conversationIds: number[]): Promise<PinConversationResponse> {
    const response = await api.put<PinConversationResponse>('/reorder', { conversationIds });
    return response.data;
  }

  static async isPinned(conversationId: number): Promise<boolean> {
    const response = await api.get<{ pinned: boolean }>(`/${conversationId}/status`);
    return response.data.pinned;
  }
}
